MuvrKit/MKWatchConnectivity.swift

```diff
@@ -145,30 +153,42 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
             let duration = to.timeIntervalSinceDate(from)
             let sampleCount = 3 * 50 * Int(duration)
             
+            // Indicates if the expected sample is in the requested range
             func isInRange(sample: CMRecordedAccelerometerData) -> Bool {
+                // check only 'start' time - don't care about end of range
                 return from.timeIntervalSince1970 <= sample.startDate.timeIntervalSince1970
             }
             
+            // Indicates if the sample is the expected one (regarding recorded time)
+            // It allows to check for ``missing`` samples in the requested range
+            func isExpectedSample(sample: CMRecordedAccelerometerData, lastTime: NSDate?) -> Bool {
+                if lastTime == nil { // first sample check it is in range
+                    return isInRange(sample)
+                } else { // check sample is not more than 40ms apart from last one
+                    return sample.startDate.timeIntervalSinceDate(lastTime!) < 0.4
+                }
+            }
+            
             if simulatedSamples {
                 let samples = (0..<sampleCount).map { _ in return Float(0) }
                 return try! MKSensorData(types: [.Accelerometer(location: .LeftWrist)], start: from.timeIntervalSince1970, samplesPerSecond: 50, samples: samples)
             } else {
+                var sampleStart: NSDate? = nil
+                var lastTime: NSDate? = nil
                 return recorder.accelerometerDataFromDate(from, toDate: to).flatMap { (recordedData: CMSensorDataList) -> MKSensorData? in
                     let samples = recordedData.enumerate().flatMap { (_, e) -> [Float] in
-                        if let data = e as? CMRecordedAccelerometerData where isInRange(data) {
+                        if let data = e as? CMRecordedAccelerometerData where isExpectedSample(data, lastTime: lastTime) {
+                            if sampleStart == nil { // first sample - set range start date
+                                sampleStart = data.startDate
+                            }
+                            lastTime = data.startDate
                             return [Float(data.acceleration.x), Float(data.acceleration.y), Float(data.acceleration.z)]
                         }
-                        NSLog("Received data outside of range: \(e)")
+                        NSLog("Received unexpected sample: \(e)")
```

Comment:
Remove that log, it generates thousands of logs when a block of data outdated is received, it even makes my app crashes sometimes